ouroboros-network-framework/src/Ouroboros/Network/Protocol/Handshake/Type.hs

```diff
@@ -111,11 +121,21 @@ instance Protocol (Handshake vNumber vParams) where
     exclusionLemma_NobodyAndClientHaveAgency TokDone    tok = case tok of {}
     exclusionLemma_NobodyAndServerHaveAgency TokDone    tok = case tok of {}
 
+-- $simultanous-open
+--
+-- On simultanous open both sides will send `MsgProposeVersions`, which will be
+-- decoded as `MsgProposeVersions'` which is a terminal message of the
+-- protocol.  It is important to stress that in this case both sides will make
+-- the choice which version and parameters to pick.  Our algorithm for picking
+-- version is symmetric, which ensures that both sides will endup with the same
+-- choice.  If one side decides to refuse the version it will close the
+-- connection, without sending the reason to the other side.
```

Comment:
TODO: 

* add a property test for the function that picks the version,  it must be symmetric,
* document this for 3rd parties.

Reply:
Elaborate on the consequences of non injective encoding and reason why this is ok here.

Reply:
done